packages/console/src/components/PlanGate.tsx

```diff
@@ -2,65 +2,70 @@
 
 import { ReactNode } from 'react'
 import { useW3 } from '@storacha/ui-react'
-import StripePricingTable, { StripeTrialPricingTable } from './PricingTable'
+import StripePricingTable, { StripeTrialPricingTable, SSOIframeStripePricingTable } from './PricingTable'
 import { TopLevelLoader } from './Loader'
 import { Logo } from '@/brand'
 import { usePlan } from '@/hooks'
 import { useRecordRefcode } from '@/lib/referrals/hooks'
 import { useSearchParams } from 'next/navigation'
+import { useIframe } from '@/contexts/IframeContext'
 
 const PricingTable = ({
   email,
   referredBy,
 }: {
   email: string
   referredBy?: string
-}) => (
-  <div className="flex flex-col justify-center items-center min-h-screen">
-    <div className="my-6">
-      <Logo />
+}) => {
+  const { isIframe } = useIframe()
+  
+  return (
+    <div className="flex flex-col justify-center items-center min-h-screen">
+      <div className="my-6">
+        <Logo />
+      </div>
+      <div className="max-w-screen-lg font-epilogue text-black text-center bg-white border border-hot-red rounded-2xl overflow-hidden p5 mx-4 mb-4">
+        {referredBy ? (
+          <>
+            <div className="px-6 py-6 lg:px-24">
+              <h1 className="my-4 font-bold">Welcome, {email}!</h1>
+              <p className="my-4">
+                Congratulations! You are eligible for a free trial of our Lite or
+                Business subscriptions. That means we won&apos;t charge you
+                anything today. If you choose a Lite plan, you will get two months
+                for free! If you choose Business, you will get one month for free!
+                We do need you to provide a valid credit card before we can start
+                your trial - pick a plan below and complete the checkout flow to
+                get started!
+              </p>
+              <p className="my-4">
+                Please note that after your free trial ends, you will be charged
+                10 USD per month for Lite or 100 USD per month for Business tier.
+              </p>
+            </div>
+            {isIframe ? <SSOIframeStripePricingTable /> : <StripeTrialPricingTable />}
```

Comment:
@travis - can we use the same pricing table when we have a `referredBy` and it is in `iframe` mode?

Reply:
oh hm, if we want to support free trials with iframe mode we'll need yet another pricing table that a) does not redirect and b) has free trials enabled.

I'm not sure free trials + iframe is necessary right now - we should confirm with @prodalex but I think we can probably punt on this for now

Reply:
(in which case I would recommend just doing this
```suggestion
            <StripeTrialPricingTable />
```